### ISSUE TITLE
Protecting config.txt on RaspberryPi?

### ISSUE THREAD

**Issue by harryberlin** [osmc/osmc#561]
Is there a way to protect the config.txt on Raspberry Pi?
In some cases by turning power off (pi is not shut down), the config.txt gets empty 0KB file.
So, on next power on, the Raspberry don't boot up.
Could it be to set the file(s) as "read only" like LibreElec to solve the issue?

**Comment by samnazarko** [comment]
The best way would be to make a backup. The card shouldn't be getting corrupted regularly, so I would verify the card.

LibreELEC doesn't use a read-only FAT32 partition, or updates would be impossible. I believe that they use squashfs for their root filesystem, but the boot files themselves are indeed writable. It's possible they mount the boot partition as RO unless there is an update action occurring.

**Comment by harryberlin** [comment]
yes, at the moment i have placed a config.txt.bak in the same path. but this needs some steps (removing card,  boot up an other pc, put card into adapter, restore the file, put card back in pi).

in LE it's mounted as ro. 
for changes you can remount as rw. after this, remount as ro.

**Comment by samnazarko** [comment]
You could mount it ro in /etc/fstab and change this on each update, but it seems a bit cumbersome. Unfortunately the current My OSMC implementation wouldn't make it trivial for us to make it RO by default.

**Comment by harryberlin** [comment]
tried to figure out, where is the sub process to place before and after these commands in:
my.osmc/script.module.osmcsetting.updates/resources/lib/update_service.py
or 
my.osmc/script.module.osmcsetting.updates/resources/lib/apt_cache_action.py
`mount -o remount,rw /boot`
#do update
`mount -o remount,ro /boot`

is there a hidden background process to run the update?

my.osmc/script.module.osmcsetting.pi 
my.osmc/script.module.osmcsetting.pioverclock
are more easy to find the step.

**Comment by samnazarko** [comment]
Be careful: we need a writable /boot for updating the kernel and firmware too.

On 15 Jun 2020, at 18:32, harryberlin <notifications@github.com> wrote:

﻿

tried to figure out, where is the sub process to place before and after these commands in:
my.osmc/script.module.osmcsetting.updates/resources/lib/update_service.py
or
my.osmc/script.module.osmcsetting.updates/resources/lib/apt_cache_action.py
mount -o remount,rw /boot
#do update
mount -o remount,ro /boot

is there a hidden background process to run the update?

my.osmc/script.module.osmcsetting.pi
my.osmc/script.module.osmcsetting.pioverclock
are more easy to find the step.

**Comment by harryberlin** [comment]
yes, i understood.
is there a process before and after updating, to placew the commands for rw and ro?

**Comment by samnazarko** [comment]
No - but I’m happy to accept a pull request to change this behaviour.

On 15 Jun 2020, at 18:51, harryberlin <notifications@github.com> wrote:

﻿

yes, i understood.
is there a process before and after updating, to placew the commands for rw and ro?
